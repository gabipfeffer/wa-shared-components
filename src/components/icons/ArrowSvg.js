// @flow
import React from 'react';
import cs from 'classnames';

import s from './Arrow.pcss';

const arrowPath =
  'M317.256 354.744l-90.512 90.512 285.256 285.254 285.254-285.256-90.508-90.508-194.746 194.744z';

export default function ArrowSvg({ color, up, id, classNames }) {
  return (
    <i
      className={cs(s.arrow, { [s.up]: up, [s.down]: !up, classNames: classNames })}
      style={{ color }}
      id={id}>
      <svg
        id={id}
        xmlns="http://www.w3.org/2000/svg"
        width="34"
        height="34"
        fill={color ? color : 'currentColor'}
        viewBox="0 0 1024 1024">
        <path id={id} d={arrowPath} />
      </svg>
    </i>
  );
}

ArrowSvg.defaultProps = {
  up: false
};
