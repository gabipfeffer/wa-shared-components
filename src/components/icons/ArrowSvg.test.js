import React from 'react';
import ArrowSvg from './ArrowSvg';

describe('ArrowSvg component', () => {
  it('render correctly with default props', () => {
    const wrapper = shallow(<ArrowSvg />);
    expect(wrapper).toMatchSnapshot();
  });

  it('is created with red color and up true', () => {
    const wrapper = shallow(<ArrowSvg up color={'red'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
