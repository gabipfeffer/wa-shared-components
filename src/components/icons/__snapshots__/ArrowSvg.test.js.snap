// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowSvg component is created with red color and up true 1`] = `
<i
  className="arrow up"
  style={
    Object {
      "color": "red",
    }
  }
>
  <svg
    fill="red"
    height="34"
    viewBox="0 0 1024 1024"
    width="34"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M317.256 354.744l-90.512 90.512 285.256 285.254 285.254-285.256-90.508-90.508-194.746 194.744z"
    />
  </svg>
</i>
`;

exports[`ArrowSvg component render correctly with default props 1`] = `
<i
  className="arrow down"
  style={
    Object {
      "color": undefined,
    }
  }
>
  <svg
    fill="currentColor"
    height="34"
    viewBox="0 0 1024 1024"
    width="34"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M317.256 354.744l-90.512 90.512 285.256 285.254 285.254-285.256-90.508-90.508-194.746 194.744z"
    />
  </svg>
</i>
`;
